src/14_bc_connectivity_data.R

```diff
@@ -0,0 +1,668 @@
+#
+# ***
+#
+#   ### **UBF RRS**
+#
+#   Refers to the **Universal Broadband Fund – Rapid Response Stream**, a fast-track funding stream for **shovel-ready projects** that can be deployed quickly in areas with urgent connectivity needs. It prioritizes projects that can deliver immediate improvements, often in rural or Indigenous communities. [\[ised-isde.canada.ca\]](https://ised-isde.canada.ca/site/high-speed-internet-canada/en/universal-broadband-fund), [\[www.advintive.com\]](http://www.advintive.com/ubf-rrs/)
+#
+# ***
+#
+# ------------------
+
+# Packages
+library(forcats)
+# --- Aggregate NBD-PHH-Speeds to Census Subdivision (CSD) level ---
+# Packages
+library(sf)
+library(dplyr)
+library(readr)
+library(stringr)
+library(tidyr)
+library(lubridate)
+library(glue)
+library(datadictionary)
+source("./src/utils.R") # get the functions for plotting maps
+
+# Load configuration using config package
+# This will automatically look for a file named config.yml in the current and parent directory
+config <- config::get()
+# use this config <- config::get(config = "development" ) to switch to development environment.
+
+# Extract settings from config
+lan_path <- config$lan_path
+connectivity_data_path <- config$conectivity_data_path
+
+
+########################################################################
+# CSD info 2021 Dissemination Geographies Relationship File (DGRF) from Statistics Canada.
+# https://www12.statcan.gc.ca/census-recensement/2021/geo/sip-pis/dguid-idugd/index2021-eng.cfm?year=21
+# or https://www150.statcan.gc.ca/n1/pub/92-150-g/92-150-g2021001-eng.htm
+
+# -------------------------------------------------------------
+# Aggregate CURRENT NBD PHH speeds (BC) to DA and CSD via DGRF
+# Date: Sys.Date()
+# -------------------------------------------------------------
+
+# -----------------------------
+# 1) CONFIG — file paths
+# -----------------------------
+
+# A) PHH speeds (Current) for BC
+path_phh_current_csv <- file.path(
+  lan_path,
+  connectivity_data_path,
+  "NBD_PHH_Speeds",
+  "PHH_Speeds_Current-PHH_Vitesses_Actuelles_BC.csv"
+)
+
+# B) PHH21 demographic points (BC) — needed for DBUID_Ididu and optional weights
+#    Expected columns from dictionary: PHH_ID, DBUID_Ididu, Pop2021, TDwell2021_TLog2021, URDwell2021_RH2021
+path_phh21_points_csv <- file.path(
+  lan_path,
+  connectivity_data_path,
+  "PHH_2021_CSV\\PHH_2021_CSV\\PHH-BC.csv"
+)
+
+# C) 2021 DGRF (Dissemination Geographies Relationship File) — nationwide CSV
+#    This links DBUID -> DAUID -> CSDUID (and others). We'll filter PRUID=59 for BC.
+
+DGRF_url <- "https://www12.statcan.gc.ca/census-recensement/2021/geo/sip-pis/dguid-idugd/files-fichiers/2021_98260004.zip"
```

Comment:
Suggestion for future is that we should probably clean up where we grab all of our DBUIDs/etc. from. I think right now we are pulling from many various sources at an ad-hoc level, depending on the script. Should probably have one source of truth for all of the DBs/DAs/CSDs/etc? Preferably from the data catalogue if we can.